astroid/nodes/node_classes.py

```diff
@@ -612,6 +612,19 @@ class AssignName(
 
     _other_fields = ("name",)
 
+    assigned_stmts: Callable[
+        [
+            Union["AssignName", "AssignAttr"],
```

Comment:
```suggestion
            "AssignName",
```
The first argument here will always be `AssignName` or a subclass, so this should work. The function type is more permissive.

Reply:
Ah, didn't think of this!